### ISSUE TITLE
We need a pattern to flow

### ISSUE THREAD

**Issue by 2shrestha22** [rrousselGit/riverpod#3499]
**Describe what scenario you think is uncovered by the existing examples/articles**
Riverpod is too flexible and developers often get lost in it. Developers needs a pattern or structure to flow so that the project remains organized. If coming from BloC, MVC or MVVM pattern developers often gets confused on what would be a better way to organize the code. Without official documentation and recommended pattern it always feels something is missing.

**Describe why existing examples/articles do not cover this case**
Official examples does not show a specific pattern to flow.

**Additional context**
Add any other context or screenshots about the documentation request here.

**Comment by snapsl** [comment]
Take a look at the third-party examples. These show programming patterns that you can follow using riverpod.

**Comment by 2shrestha22** [comment]
Third party examples doesn't solve the problem. People tends to create their own solution. Let's take bloc as an example. There is clear instructions on how you name your bloc, state and events and where you put bloc and state. Bloc always feels structured than riverpod. Our team is not moving forward with riverpod because of this. I feel I don't know how to describe it.

**Comment by snapsl** [comment]
I agree that the bloc documentation is more comprehensive and mature than the current state of the riverpod documentation. Also, bloc comes with its own architecture pattern (BLoC). Riverpod, on the other hand, is a state managment framework (docs say a "reactive caching framework"). It does not force you to use a specific architecture pattern such as BloC, MVC or MVVM, but you can choose one. That is why it is not part of the documentation and probably never will be.
